keel-sql/src/main/kotlin/com/netflix/spinnaker/jooq/enumConverters.kt

```diff
@@ -0,0 +1,63 @@
+package com.netflix.spinnaker.jooq
+
+import com.netflix.spinnaker.keel.api.artifacts.ArtifactStatus
+import com.netflix.spinnaker.keel.api.constraints.ConstraintStatus
+import com.netflix.spinnaker.keel.api.verification.VerificationStatus
+import com.netflix.spinnaker.keel.core.api.PromotionStatus
+import com.netflix.spinnaker.keel.events.PersistentEvent
+import com.netflix.spinnaker.keel.lifecycle.LifecycleEventStatus
+import com.netflix.spinnaker.keel.lifecycle.LifecycleEventType
+import com.netflix.spinnaker.keel.notifications.NotificationScope
+import com.netflix.spinnaker.keel.notifications.NotificationType
+import com.netflix.spinnaker.keel.pause.PauseScope
+import org.jooq.impl.EnumConverter
+
+class ArtifactStatusConverter : EnumConverter<String, ArtifactStatus>(
```

Comment:
Too bad we can't do this with a generic type to avoid the duplication. Oh well.

Reply:
We could if the XML would allow for method references. There is a configuration DSL but I think we'd get into circular dependency issues because we're using the code generation stuff.

Reply:
It's alright, this is a great improvement as it is.